src/components/atoms/Navigator/index.tsx

```diff
@@ -0,0 +1,177 @@
+import { memo } from "react";
+
+import { useT } from "@reearth/i18n";
+import { styled } from "@reearth/theme";
+
+import Icon from "../Icon";
+
+import { useNavigator } from "./hooks";
+import { CompassAngle } from "./types";
+
+export type Props = {
+  degree: number;
+  /**
+   * Pass degree of circle as callback arguments.
+   * This event is invoked when outer circle is clicked and mouse move.
+   * If click is dismissed then this event is stopped.
+   */
+  onRotate?: (degree: number) => void;
+  /**
+   * Pass distance of movement as x,y axis(px) and degree of circle as callback arguments.
+   * This event is invoked when inner circle button is clicked and mouse move.
+   * If click is dismissed then this event is stopped.
+   */
+  onOrbit?: (angle: CompassAngle) => void;
+  /**
+   * This event is invoked when center of circle is clicked.
+   */
+  onRestoreRotate?: () => void;
+  /**
+   * Pass + or -  as callback arguments.
+   */
+  onClickHelp?: () => void;
+  onZoomIn?: () => void;
+  onZoomOut?: () => void;
+};
+
+const Navigator: React.FC<Props> = memo(function NavigatorPresenter({
+  degree,
+  onRotate,
+  onOrbit,
+  onRestoreRotate,
+  onClickHelp,
+  onZoomIn,
+  onZoomOut,
+}) {
+  const {
+    compassRef,
+    compassDegree,
+    compassFocusAngle,
+    isMovingAngle,
+    handleOnMouseDownAngle,
+    handleOnMouseDownCompass,
+  } = useNavigator({ degree, onRotate, onOrbit });
+  const t = useT();
+  return (
+    <Container>
+      <CompassContainer>
+        <Compass ref={compassRef}>
+          <>
+            <CompassIcon onMouseDown={handleOnMouseDownCompass}>
+              <Icon
+                icon="compass"
+                color="#000"
+                aria-label={t("compass")}
+                size={64}
+                style={{ transform: `rotate(${compassDegree}deg)` }}
+              />
+            </CompassIcon>
+            {isMovingAngle && (
+              <CompassFocusIcon
+                style={{
+                  transform: `rotate(${compassFocusAngle.degree}deg)`,
+                }}
+                data-testId="compassFocus">
+                <Icon icon="compassFocus" color="blue" alt="" size={30} />
+              </CompassFocusIcon>
```

Comment:
Same as the comment above. You can resolve this comment if it seems appropriate.